src/app/modules/submission/encrypt-submission/encrypt-submission.service.ts

```diff
@@ -530,3 +534,31 @@ export const performEncryptPostSubmissionActions = (
       })
     })
 }
+
+export const getQuarantinePresignedPostData = (
+  attachmentSizes: Record<string, number>,
+): Result<Record<string, PresignedPost>, unknown> => {
+  const attachmentPresignedData: Record<string, PresignedPost> = {}
+  try {
+    for (const [id, contentLength] of Object.entries(attachmentSizes)) {
+      if (!mongoose.isValidObjectId(id)) return err(new InvalidFieldIdError())
+      const presignedPostData = AwsConfig.s3.createPresignedPost({
+        Bucket: AwsConfig.virusScannerQuarantineS3Bucket,
+        Fields: { key: crypto.randomUUID() },
+        Expires: 5 * 60, // expires in 5 minutes
```

Comment:
Suggest 1 minute (the minimum) to reduce risk of client reusing the link maliciously - the client just needs to initiate the upload within 1 minute, does not need to complete within the 1 minute

(see [link](https://docs.aws.amazon.com/AmazonS3/latest/userguide/using-presigned-url.html): Amazon S3 checks the expiration date and time of a signed URL at the time of the HTTP request. For example, if a client begins to download a large file immediately before the expiration time, the download continues even if the expiration time passes during the download. )

Reply:
Okay, yup! Done in [51ce4c2](https://github.com/opengovsg/FormSG/pull/6685/commits/51ce4c22bbec7a24c8c73a25eca7eb7dcc9cd076). Updated virus scanning design docs too.

Reply:
ah yes thanks @LinHuiqing ! i forgot to update the virus scanner docs - i discussed this with sebastian during the PR review of the virus scanner 😅😃

Reply:
sorry one more thing : at this point i think we should also check that the total attachment size does not exceed form limits. otherwise the client could upload arbitrarily large files to our bucket

Reply:
Yes, done in [447b6ab](https://github.com/opengovsg/FormSG/pull/6685/commits/447b6abfcc83cf79ee8ca232155df9bb0a15fde7) 👍